templates/purge-registry.py.j2

```
@@ -0,0 +1,142 @@
+        logging.error('Error %s when trying to delete artifact %s:%s from %s: %s' % (err.response.status_code, art.name, art.version, art.owner, err.response.text))
+        raise err
+
+def prepare_credentials(args):
+    username=None
+    password=None
+    if args.username is not None:
+        logging.info('Loading username from parameter')
+    if args.password is not None:
+        logging.warn('Loading password from parameter, not recommended')
+    if environ.get('GITEA_USERNAME') is not None:
+        logging.info('Loading username from environment variable')
+        username=environ.get('GITEA_USERNAME')
+    if environ.get('GITEA_USERNAME') is not None:
+        logging.info('Loading password from environment variable')
+        username=environ.get('GITEA_PASSWORD')
+    if username is None or password is None:
+        logging.error('Missing username or password, use environment variable GITEA_USERNAME and GITEA_PASSWORD')
+        exit(1)
+    return requests.auth.HTTPBasicAuth(username, password)
+        
+def parse_args():
+    parser = ArgumentParser(description=HELP_DEFINITION, epilog=HELP_EXAMPLE, formatter_class=RawTextHelpFormatter)
+    parser.add_argument('-r', '--retention',  required=True,  help='Maximum retention period for each artifact in days', type=int)
+    parser.add_argument('-U', '--url',        required=True,  help='URL of the Gitea instance')
+    parser.add_argument('-u', '--user',       required=False,  help='Username')
+    parser.add_argument('-p', '--password',   required=False,  help='Password')
+    parser.add_argument('-l', '--log',        required=False, default='INFO')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    logging.basicConfig(level=args.log.upper())
+    basic = prepare_credentials(args)
+    retention=int(args.retention)
+    orgs=[]
+    try:
+        orgs=get_organization(args.url, basic)
+    except request.exceptions.HTTPError as err:
+        exit(1)
```

What is the point of this `try`/`except`? Why are you hiding an exception? Just let it bubble to the top and break.